src/util/std_code.h

```diff
@@ -1557,18 +1557,19 @@ inline const code_expressiont &to_code_expression(const codet &code)
 /// subtypes are not subtypes of \ref codet, but they inherit directly from
 /// \ref exprt. They do have a `statement` like [codets](\ref codet), but their
 /// [id()](\ref irept::id) is `ID_side_effect`, not `ID_code`.
-class side_effect_exprt:public exprt
+class side_effect_exprt : public multi_ary_exprt
```

Comment:
I don't think that will work out: a `side_effect_expr_nondett` is a `side_effect_exprt` without operands.

Reply:
mutli_ary_exprt wasn't meant to guarantee that there's at least one operand -- however, reverting for now.

Reply:
Ok, might be best sorted out in a separate PR. We should certainly not introduce fresh use of deprecated functions, so either a different constructor needs to be used or some of the constructors of `multi_ary_exprt` need to be de-deprecated.